[TextFields] Ensures layoutDelegate is informed of size changes in the text field

felixaemiliano opened the pull request:
Removes requirement that the textView's delegate be set when informing layout delegate of a size change

Closes #9262

randallli committed:
Merge branch 'release-candidate' into stable

randallli committed:
Merge branch 'release-candidate' into stable

randallli committed:
Automatic changelog preparation for release.

randallli committed:
revert of commit ca2f2ad64046b4a2583abbe7cb705e42ed5f0ae7

randallli committed:
added changes based on commit messages.

randallli committed:
corrected changelog because of the reverted PR

randallli committed:
first pass raw adding API diff

randallli committed:
refined API diff

randallli committed:
Bumped version number to 97.0.0.

randallli committed:
Revert "Delete MDCButtonBarColorThemer (#9235)"

This reverts commit d0c581e993a7de09c95a2b6c067733fed0239c0b.

randallli committed:
removed change long entries for buttonBar

felixaemiliano committed:
Removes requirement that the textView delegate be set when informing layout delegate of a size change

andrewoverton reviewed (approved):
Merge conflicts but LGTM

felixaemiliano committed:
Merge branch 'develop' into textfielddelegate

randallli committed:
Merge branch 'release-candidate' into stable

randallli committed:
merged release-candidate

Commit:
[Chips] Fix `contentPadding` API. (#9264)

Corrects the behavior of `contentPadding` to ensure that it correctly
positions content within its padding bounds.

This change includes two corrections that addressed problems in layout:

*   Calculating subview bounding rectangles within the `MDCChipView` failed to account for `contentPadding.top` when determining the bounding rectangle origin.
*   Positioning the subviews vertically within the bounding rectangle was accidentally using the `MDCChipView` frame rather than the content rectangle. This resulted in the content padding being evenly "split" between the top and bottom rather than the actual values.

Closes #9249

Commit:
[Chips] Add snapshot tests for selectedImage layout. (#9266)

Adds more snapshot tests to capture the behavior of Chips when a
`selectedImageView.image` is set. An upcoming change will correct the layout
of subviews when `imagePadding` is used, and these tests capture errors that
take place when a `selectedImageView.image` is set.

Preparation for #9250

randallli committed:
[Cards] Removed podspec for card themer because it was removed in #9197 (#9268)

The code changes happened: 
https://github.com/material-components/material-components-ios/pull/9197

https://github.com/material-components/material-components-ios/commit/f44e5d679d78c01957677ee04c0471c57f0cad4b
(cherry picked from commit e9f1ac8e3b82774295e96ac6d8445d5c25d0a328)

randallli committed:
[Cards] Removed podspec for card themer because it was removed in #9197 (#9268)

The code changes happened: 
https://github.com/material-components/material-components-ios/pull/9197

https://github.com/material-components/material-components-ios/commit/f44e5d679d78c01957677ee04c0471c57f0cad4b

randallli committed:
Automatic changelog preparation for release.

randallli committed:
Bumped version number to 97.0.1.

randallli committed:
Hand-modified CHANGELOG.md API diff.

Commit:
[Chips] Fix `imagePadding` behavior. (#9267)

Corrects the behavior of `imagePadding` in two significant ways:

*   Previously, the union of the frames of `selectedImageView` and `imageView`
    were used to determine the position of the `titleLabel`. However, since
    `selectedImageView` is documented as not having an effect on
    `MDCChipView` unless it is set and the Chip is selected, it should not have
    any effect on the frame calculations when unselected or unset.
*   `imagePadding` is now only applied if either `selectedImageView` or
    `imageView` in the Chip. Previously, `imagePadding` would have been
    applied even without an image present.

Closes #9250

Commit:
[BottomNavigation] Use NotoNastaliq for Arabic/RTL tests. (#9260)

NotoNastaliq is a font with a greater line height than most other fonts. Using
it for Arabic/RTL fonts results in greater exercise of the view's layout.

Part of #6552

bryanoltman committed:
[Cards] Remove references to Themer from README (#9261)

Card Themer classes were deleted, and our documentation should be updated to reflect that.

Fixes #9197

randallli committed:
merged release-candidate

randallli committed:
Release cut adjustment to generated change log (#9221)

Removed `## Changes` heading from the auto generated changes in release cut script because heading is a duplicate of the one above it.

galiak11 committed:
[List] Deprecate MDCListTypographyThemer (#9269)

Deprecate symbol MDCListTypographyThemer (Issue #9222)

galiak11 committed:
[List] Deprecaet MDCListColorThemer (#9265)

Deprecating MDCListColorThemer (b/145204461).

randallli committed:
removed build file dep that was deleted but the colorThemer revert accidentally readd it (#9278)

TabFontThemer  that was deleted but the colorThemer revert accidentally readd it

Here is the botched revert:
https://github.com/material-components/material-components-ios/commit/5de47f883ca7be62c7ed11ac373bf04d7886ab75#diff-eac85c81daeac71b652d1a98c5bf5bbc

randallli committed:
Automatic changelog preparation for release.

randallli committed:
Hand-modified CHANGELOG.md API diff.

randallli committed:
Hand-modified CHANGELOG.md API diff.

bryanoltman committed:
[FlexibleHeader] Deprecate MDCFlexibleHeaderColorThemer (#9281)

MDCFlexibleHeaderColorThemer was marked as "ToBeDeprecated". This PR performs that deprecation.

bryanoltman committed:
[FlexibleHeader] Remove uses of MDCFlexibleHeaderColorThemer (#9282)

Removes our use of the newly-deprecated `MDCFlexibleHeaderColorThemer`.

Part of:
#9284
#9285
#9286
#9287

bryanoltman committed:
[BottomAppBar] Add snapshot test to verify autolayout bug (#9289)

* [BottomAppBar] Add snapshot test to verify autolayout bug

* Update snapshot

bryanoltman committed:
[BottomAppBar] Add intrinsicContentSize (#9290)

* [BottomAppBar] Add snapshot test to verify autolayout bug

* Update snapshot

* [BottomAppBar] Add intrinsicContentSize

andrewoverton committed:
Revert "Revert "Delete MDCButtonBarColorThemer (#9235)"" (#9272)

This PR brings back https://github.com/material-components/material-components-ios/pull/9235, which was reverted because [cl/285188391](http://cl/285188391) had not actually been submitted when #9235 was merged. It is now safe to merge.

andrewoverton committed:
Delete MDCOutlinedTextFieldColorThemer (#9274)

BUILD and Podspec changes coming with the deletion of MDCFilledTextFieldColorThemer.

Closes #9080.

andrewoverton committed:
Delete MDCFeatureHighlightTypographyThemer (#9273)

This PR deletes MDCFeatureHighlightTypographyThemer.

Closes #9208.

andrewoverton committed:
Revert "revert of commit ca2f2ad64046b4a2583abbe7cb705e42ed5f0ae7" (#9271)

This PR brings back https://github.com/material-components/material-components-ios/pull/9200. It was reverted because there were some remaining references to MDCTabBarColorThemer. [cl/284979182](http://cl/284979182) addresses those remaining references. This PR should be merged after the CL is approved and submitted.

williamernest committed:
[Chips] Fix documentation error for Theming section (#9294)

bryanoltman committed:
[Cards] Remove leading space before Swift and ObjC example headers (#9295)

Removes leading spaces before "Swift" and "Objective C" to ensure proper layout on material.io/develop/ios/components/cards (related to https://github.com/material-components/material-components-ios/pull/9294)

galiak11 committed:
[List] Remove MDCListColorThemer (#9188)

* Removing MDCListColorThemer

* Removing ColorThemer targets from BUILD file

randallli committed:
minor edit to changelog

bambaragoogle committed:
[Dialogs] Fix for MDCAlertController icon centering. (#9219)

Title icons for MDCAlertController should mimic the alignment of the title. When centering the title on an alert, the icon is not correctly centered (Issue: #9216)

randallli committed:
Revert "[Chips] Fix `imagePadding` behavior. (#9267)"

This reverts commit 7a5cc0f17a114a3dfa1301048d6410ce89d0a317.

randallli committed:
Revert "[Chips] Add snapshot tests for selectedImage layout. (#9266)"

This reverts commit c9cc0f1a6bf7dc9e92b758e7783e48606ee9b1b7.

randallli committed:
Revert "[Chips] Fix `contentPadding` API. (#9264)"

This reverts commit 0f603de7010ec2c444c265ac54c2cc465cbeeb10.

randallli committed:
Revert "[Chips] Tests to evaluate `padding` APIs. (#9252)"

This reverts commit 2bd1b89c88d505fadbe45f7a0c21534802e8c2d0.

randallli committed:
Hand-modified CHANGELOG.md API diff.

randallli committed:
Bumped version number to 98.0.0.

randallli committed:
merged release-candidate

galiak11 committed:
[List] Removing Themers from documentation (#9296)

Removing MDCList Themers from documentation, replacing them with theming extensions. (b/145204420, b/145204461)

yarneo committed:
[Snackbar] Allow Snackbars to be non-transient if needed. (#9299)

This provides an opt-in functionality for clients to have the Snackbar persist unless an action is made upon it.

galiak11 committed:
[List] Remove MDCListTypographyThemer (#9190)

Delete symbol MDCListTypographyThemer from our library (b/145204420).

galiak11 committed:
[List] Fix typo in doc (#9301)

Fix typo in MDCList readme.

Commit:
[scripts/release] Always branch off of stable. (#9310)

Prior to this change, our release process was merging `develop` directly into `stable` without first branching off of `stable`.

The result of this process was that `stable` shares no common ancestor with `develop`, making it difficult to calculate differences between a commit on `develop` and a commit on `stable`.

One example of where this becomes problematic is in generating internal CLs for our PRs and for `develop`. It is not possible to compute the delta between the most recent `stable` commit that was mirrored internally and a given `develop` commit, making it difficult for tooling to apply deltas to the internal codebase without simply blowing away the internal directory and checking out the desired `develop` commit.

To fix this problem, we need to create a common ancestor between `develop` and `stable`. To do so, when cutting a release we need to cut the `release-candidate` from `stable` rather than `develop`, and then immediately merge `develop` into the release-candidate. This creates a common ancestor between `develop` and `stable`.

Note that this change causes us to deviate from the pure git flow model we have historically been following, as shown in the before/after of the model below (note the new red lines connecting stable to develop).

| Before | After |
|:--|:--|
| ![git-model@2x](https://user-images.githubusercontent.com/45670/71178792-ebf05600-223c-11ea-9f16-df72bd13a067.png) | ![git-model@2x copy 2](https://user-images.githubusercontent.com/45670/71179290-dd566e80-223d-11ea-8eab-2487490985f6.png) |

yarneo committed:
[Snackbar] Add snapshot test showing reported shadow bug (#9309)

This snapshot test for the Snackbar exposes the bug as seen in https://github.com/material-components/material-components-ios/issues/9308

bryanoltman committed:
[Dialogs] Expose MDCAlertController presentation animation properties (#9314)

Exposes MDCDialogTransitionController scaleAnimationDuration, opacityAnimationDuration, and dialogInitialScaleFactor presentation configuration properties through MDCAlertController. 

Fixes #9313

yarneo committed:
[Snackbar] Resolve truncation of shadow in Snackbars (#9315)

With this change we are allowing the Snackbar shadow to go beyond the layout constraints of the underlying SnackbarOverlayView that prior caused a weird truncation of shadow.

Closes #9308 

Yay snapshots!

yarneo committed:
Resolve kokoro build failure for CocoaPods builds. (#9306)

this PR uninstalls the rubygems-bundler which is not needed anymore and causing build failures. See more details in tracking bug https://github.com/material-components/material-components-ios/issues/9305

Commit:
Clean up unsupported situation when inferring the top safe area inset (#9317)

In the case of automatically selecting a view controller, when the selected view controller is self.topLayoutGuideViewController, self.topLayoutGuideViewController.parentViewController is guaranteed to be nil, and [self fhv_rootAncestorOfViewController:nil] returns nil. Therefore, just set ancestor = nil directly.

An argument could be made that the same check for self.topLayoutGuideAdjustmentEnabled && ancestor == self.topLayoutGuideViewController should be made when the delegate returns a value, but that would result in a behavior change (infinitely increasing the top layout guide vs immediately crashing).

PiperOrigin-RevId: 283419359

randallli committed:
[UIApplication] Use prefix on category name for UIApplication+AppExtensions so that category names don't conflict in ObjC namespace. (#9277)

[UIApplication] Use prefix on category name for UIApplication+AppExtensions so that category names don't conflict in ObjC namespace.

sourced from cl/285233749

yarneo committed:
[Ink] Update ink color to be null resettable. (#9253)

This allows the API to handle assigning inkColor to nil, which is common practice among properties on UI classes that expose color (see UIKit - i.e. UIView.tintColor, UILabel.textColor, etc)

b/144862050

andrewoverton committed:
Remove out of date color theming docs (#9330)

This PR removes out of date color theming docs. Forgotten cleanup related to #9061.

material-automation commented:
bazel detected changes to the following targets:

```
//components/Chips:snapshot_tests
//components/Chips:snapshot_tests_test_bundle
//components/Chips:unit_tests
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Chips:unit_tests_environment
//components/Chips:unit_tests_environment_test_bundle
//components/TextFields:snapshot_tests
//components/TextFields:snapshot_tests_test_bundle
//components/TextFields:unit_tests
//components/TextFields:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/TextFields:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/TextFields:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/TextFields:unit_tests_IPHONE_X_IN_11_0
//components/TextFields:unit_tests_IPHONE_X_IN_11_0_test_bundle
```

<!-- identifier: affected-targets -->

material-automation commented on components/TextFields/src/MDCMultilineTextField.m:
```diff
@@ -755,9 +755,9 @@ - (void)textViewDidChange:(__unused NSNotification *)note {
-                                                      didChangeContentSize:)]) {
+  if (currentSize.height != requiredSize.height && self.layoutDelegate &&
+      [self.layoutDelegate
+          respondsToSelector:@selector(multilineTextField:didChangeContentSize:)]) {
```
clang-format suggested the following change:


```
-      [self.layoutDelegate
-          respondsToSelector:@selector(multilineTextField:didChangeContentSize:)]) {
+      [self.layoutDelegate respondsToSelector:@selector(multilineTextField:
+                                                      didChangeContentSize:)]) {
```

<!-- identifier: clang-format -->
<!-- File: components/TextFields/src/MDCMultilineTextField.m -->

material-automation commented:
clang-format recommended changes.

Consider installing [git-clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) and running the following command:

```bash
git clang-format $(git merge-base origin/develop HEAD)
```

<!-- identifier: clang-format -->

Commit:
[ActionSheet] Standardize the examples. (#9334)

- Examples do not need .h files, so those have been removed.
- Example filenames should match the format "<Component><Example>Example.(h|swift)". The Swift example has been clarified as a "Typical Use" example accordingly.

Conventions: https://github.com/material-foundation/material-foundation-conventions-objc

felixaemiliano committed:
Merge remote-tracking branch 'origin/textfielddelegate' into textfielddelegate

andrewoverton commented:
These changes made it in with https://github.com/material-components/material-components-ios/pull/9337
